TextParser/Parser.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TextParser
+{
+    public class Parser
+    {
+        private List<CityAtributes> _cities;
+        private string _input;
+        private const char CitiesSeparator = ',';
+        private const char NameAndPopulationSeparator = '=';
+
+        /// <summary>
+        /// Default Constructor(Empty, use set input string method)
+        /// </summary>
+        public Parser()
+        {
+        }
+
+        /// <summary>
+        /// Input String 
+        /// </summary>
+        /// <param name="input"></param>
+        public Parser(string input)
+        {
+            this._input = input;
+        }
+
+        /// <summary>
+        /// Method Set Input string (if you yse default constructor)
+        /// </summary>
+        /// <param name="input"></param>
+        public void SetInputString(string input)
+        {
+            if (input == String.Empty)
+            {
+                throw new ArgumentNullException();
+            }
+            this._input = input;
+        }
+
+        private void Parse()
+        {
+            this._cities = new List<CityAtributes>();
+            var SplittedCities = _input.Split(CitiesSeparator);
+            //string.Join(NameAndPopulationSeparator, SplittedCities);
+            var Cities = GetCitiesNames(SplittedCities);
+            var Population = GetPopulations(SplittedCities);
+            for (int i = 0; i < Cities.Length; i++)
```

Comment:
I think you'd better go by cityAndPopulation here (just split input by ',' and then split by '=' in the loop and return 'new City (cityAndPopulation[0], cityAndPopulation[1])'